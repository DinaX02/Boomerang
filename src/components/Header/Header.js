import React from "react";
import arrowBack from "../../assets/icons/back_arrow.svg"
import "./../components.css";
import { useNavigate } from "react-router-dom";

const Header = (props) => {
  const navigate = useNavigate();
  let name = props.name;

  if (!props.name) {
    name = "Página desconhecida"
  }

  const handleGoBack = () => {
    props.alertHandler ? props.alertHandler() : navigate(-1); // voltar para trás
  };

  return (
    <header className="headerBoomerang">
      <img data-testid="svg-icon" src={arrowBack} onClick={handleGoBack} style={{ cursor: "pointer" }} alt="arrow go back" />
      <h3>{name}</h3>
    </header>
  );
};

export default Header;
