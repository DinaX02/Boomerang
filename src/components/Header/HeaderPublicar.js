import React from "react";
import arrowBack from "../../assets/icons/back_arrow.svg";
import "./../components.css";
import { useNavigate } from "react-router-dom";

const HeaderPublish = (props) => {
  const navigate = useNavigate();
  let name = props.name;

  if (!props.name) {
    name = "Página desconhecida";
  }

  const handleGoBack = () => {
    props.alertHandler ? props.alertHandler() : navigate("/"); // voltar para hompepage
  };

  return (
    <header className="headerBoomerang">
      <img
        data-testid="svg-icon"
        src={arrowBack}
        onClick={handleGoBack}
        style={{ cursor: "pointer" }}
        alt="seta para voltar à página anterior"
      />
      <h1 style={{fontSize: "16px", marginBottom: "0"}}>{name}</h1>
    </header>
  );
};

export default HeaderPublish;
