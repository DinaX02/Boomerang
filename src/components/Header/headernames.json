{
    "publicar-page": "Publicar",
    "criarConta": "Criar Conta",
    "editarPerfil": "Editar Perfil",
    "editarArtigo": "Editar Artigo",
    "artigo": "Artigo",
    "periodoDeAluguer": "Período de Aluguer",
    "detalhesDoAluguer": "Detalhes do Aluguer",
    "valorTotal": "Valor Total",
    "moradas": "Moradas",
    "adicionarMoradas": "Adicionar Moradas",
    "metodosDePagamento": "Métodos de Pagamento",
    "novoMetodoDePagamento": "Novo Método de Pagamento",
    "detalhesSobreOAluguer": "Detalhes sobre o Aluguer"
  }