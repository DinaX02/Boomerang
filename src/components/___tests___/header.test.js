import { screen, render, cleanup } from "@testing-library/react";
import pageNames from "../Header/headernames.json";
import Header from "../Header/Header";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("verifica a quantidade de pageNames estao no JSON para utilizar no Header", () => {
  let countPages = 0;

  for (let i in pageNames) {
    countPages++;
  }

  console.log(`NÃºmero de pageNames estao no JSON: ${countPages}`);

  expect(countPages).toBeGreaterThan(0);
});

test("verifica se existe um svg no componente Header", () => {
  render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );

  const svgElement = screen.getByTestId("svg-icon");

  expect(svgElement).toBeInTheDocument();
});

test("snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    .toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
