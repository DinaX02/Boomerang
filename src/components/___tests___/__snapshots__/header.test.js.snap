// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<header
  className="headerBoomerang"
>
  <img
    alt="arrow go back"
    data-testid="svg-icon"
    onClick={[Function]}
    src="back_arrow.svg"
    style={
      Object {
        "cursor": "pointer",
      }
    }
  />
  <h3>
    PÃ¡gina desconhecida
  </h3>
</header>
`;
